### PR TITLE
Implement significantly stricter linter

### PR DESCRIPTION
**Pull request by fidian**
This adds a much more strict .eslintrc (also converts it to YAML).  I've
ignored node_modules/ and coverage/, but the rest of the files now get
scanned when running "npm test".

There were many little changes that needed to get made so the linter
would not find more problems.  It's a little strict, like jslint, but
that means our code will be more uniform.

### CODE REVIEW

File: debug/cipher-info.js [hunk 1]
```diff
@@ -50,21 +69,30 @@ function scan(startKey, startIv, modKey, modIv, endKey, endIv) {
...
 }
 
-function show(result, suffix) {
-    if (! result) {
-        console.log("failureFinding" + suffix + ":", true);
+
+/**
+ * Show the results
+ *
+ * @param {Object} resultObj Object returned from scan()
```
Comment (AbsentSemicolon) [inline debug/cipher-info.js]: Should this have a typedef so we know what's contained in the object?

File: lib/encryption.js [hunk 2]
```diff
@@ -110,12 +111,77 @@ module.exports = function (ciphersAndHashes, crypto, promise, random) {
...
+    }
+
+
+    /**
+     * Reads a header and generates the parameters from the data
+     *
+     * @param {Buffer} data
+     * @return {Promise.<encryption~header>}
+     */
+    function readHeaderAsync(data) {
```
Comment (AbsentSemicolon) [inline lib/encryption.js]: If the functions need to be in alphabetical order, parametersFromHeader comes before readHeaderAsync.

Comment (fidian) [inline lib/encryption.js]: They might be, but order is now more dictated by the necessity of defining a variable or function before it is used.

**Comment by AbsentSemicolon** [comment]
Reviewed and looks good.

**Pull request merged**